### PR TITLE
Create sidebar component

### PR DESCRIPTION
**Pull request by stenizwahyudi**

### CODE REVIEW

File: src/stories/components/Sidebar/styles.scss [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    display: block;
+}
+
+/* When you mouse over the navigation links, change their color */
+a:hover {
+    color: #f1f1f1;
+}
+
+/* On smaller screens, where height is less than 450px, change the style of the sidebar (less padding and a smaller font size) */
+@media screen and (max-height: 450px) {
```
Comment (alecherryy) [inline src/stories/components/Sidebar/styles.scss:61]: We have `@mixins` for responsive breakpoints. Example usage:

```
.sidebar {
    width: 100%;
    
    @include bp-tablet {
        width: 50%;
    }
}
```

File: src/stories/components/Sidebar/Sidebar.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ * )
+ */
+
+export const Sidebar = ({ modifierClasses, sidebarContent, mainContent}) => {
+    return (
+        <div
+            className={['sidebar', `${modifierClasses}`].join(' ').trim()}
+        >
+            <div>{sidebarContent}</div>
+            <div>{mainContent}</div>
```
Comment (alecherryy) [inline src/stories/components/Sidebar/Sidebar.js:27]: Add classes to the child divs:
```
<div className="sidebar__aside">{sidebarContent}</div>
<div className="sidebar__main">{sidebarContent}</div>
```

File: src/stories/components/Sidebar/Sidebar.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import React from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * Component for sidebar element.
+ *
+ * @component
+ * @param {string} modifierClasses Class modifiers of the component.
+ * @param {html} sidebarContent Sidebar content of the component.
+ * @param {html} mainContent Main content of the component.
```
Comment (alecherryy) [inline src/stories/components/Sidebar/Sidebar.js:12]: I recently found out the correct parameter type for this is `node`.

File: src/stories/components/Sidebar/Sidebar.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+Sidebar.propTypes = {
+    /**
+   * Sidebar's modifier classes
+   */
+    modifierClasses: PropTypes.string,
+    /**
+   * Sidebar's sidebar content
+   */
+    sidebarContent: PropTypes.string,
```
Comment (alecherryy) [inline src/stories/components/Sidebar/Sidebar.js:40]: Update to `node` type.

File: src/stories/components/Sidebar/Sidebar.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    /**
+   * Sidebar's main content
+   */
+    mainContent: PropTypes.string,
```
Comment (alecherryy) [inline src/stories/components/Sidebar/Sidebar.js:44]: Update to `node` type.

File: src/stories/components/Sidebar/Sidebar.stories.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+        <Sidebar {...args} />
+    );
+};
+
+// Default Constrain
+export const Default = Template.bind({});
+Default.args = {
+    modifierClasses: '',
+    sidebarContent:
+        <div>
```
Comment (alecherryy) [inline src/stories/components/Sidebar/Sidebar.stories.js:24]: No need to double nest dummy content.

File: src/stories/components/Sidebar/Sidebar.stories.js [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+            quo no sanctus eloquentiam.</p>
+        </div>,
+};
+
+// Right Sidebar
+export const Right = Template.bind({});
+Right.args = {
+    modifierClasses: 'sidebar--right',
+    sidebarContent:
+        <div>
```
Comment (alecherryy) [inline src/stories/components/Sidebar/Sidebar.stories.js:59]: Same as above.

File: src/stories/components/Sidebar/styles.scss [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+        left: 0; /* Stay on the left side of the page */
+        background-color: #fff;
+        overflow-x: hidden; /* Disable horizontal scroll */
+        padding-top: 50px;
+    }
+    /* The main content */
+    .main {
+        margin-left: 210px; /* Same as the width of the sidebar + 50px */
+        padding: 20px 0px;
+    }
```
Comment (alecherryy) [inline src/stories/components/Sidebar/styles.scss:23]: I see what you meant to do, but we do not want to style any of the dummy content here. You're styling the content inside the Sidebar component to create the layout, which is not what we want.

File: src/stories/components/Sidebar/styles.scss [hunk 9]
```diff
@@ -0,0 +1,78 @@
+@import '../../../scss/variables.scss';
+
+// @file
+// Sidebar
+
+.sidebar {
```
Comment (alecherryy) [inline src/stories/components/Sidebar/styles.scss:6]: There are no styles apply to this. The layout should be dictated by the `sidebar` not the child content; remove the styles you have and replace them with these:

```
sidebar {
    @include bp-tablet-lg {
        display: flex;
    }
}

.sidebar__aside {
    flex-basis: rem(210px);

    @include bp-tablet-lg {
        margin-right: $spacing-3;
    }

    @include bp-desktop {
        margin-right: $spacing-5;
    }
}

.sidebar--right {
    .sidebar__aside {
        ...
        }
    }
}
```

Note that this is just to get you started. You'll have to adjust the display and styles for the `sidebar--right`.

**Review by alecherryy — CHANGES_REQUESTED** [review]

**Comment by alecherryy** [comment]
@stenizwahyudi back to you to make changes!

File: src/stories/components/Sidebar/styles.scss (hunk 1 above)
Comment (stenizwahyudi) [reply src/stories/components/Sidebar/styles.scss]: Sorry I didn't check that we have @mixins for responsive breakpoints. Noted!

File: src/stories/components/Sidebar/Sidebar.js (hunk 2 above)
Comment (stenizwahyudi) [reply src/stories/components/Sidebar/Sidebar.js]: I changed things a little bit into left content and right content for more generalization. Hope this is better.

File: src/stories/components/Sidebar/Sidebar.js (hunk 3 above)
Comment (stenizwahyudi) [reply src/stories/components/Sidebar/Sidebar.js]: Thank you!

File: src/stories/components/Sidebar/Sidebar.js (hunk 4 above)
Comment (stenizwahyudi) [reply src/stories/components/Sidebar/Sidebar.js]: Got it!

File: src/stories/components/Sidebar/Sidebar.js (hunk 5 above)
Comment (stenizwahyudi) [reply src/stories/components/Sidebar/Sidebar.js]: Sure thing!

File: src/stories/components/Sidebar/Sidebar.stories.js (hunk 6 above)
Comment (stenizwahyudi) [reply src/stories/components/Sidebar/Sidebar.stories.js:24]: Got it!

File: src/stories/components/Sidebar/Sidebar.stories.js (hunk 7 above)
Comment (stenizwahyudi) [reply src/stories/components/Sidebar/Sidebar.stories.js:38]: Got it!

File: src/stories/components/Sidebar/styles.scss (hunk 8 above)
Comment (stenizwahyudi) [reply src/stories/components/Sidebar/styles.scss]: Yes, that makes more sense. Sorry I got confused a little bit.

File: src/stories/components/Sidebar/styles.scss (hunk 9 above)
Comment (stenizwahyudi) [reply src/stories/components/Sidebar/styles.scss:6]: Awesome, thank you!

**Review by alecherryy — APPROVED** [review]

**Pull request merged**